### DISCUSSION TITLE
Universal design, and the reason why UI changes are not the focus of current development

### DISCUSSION THREAD

**Discussion by JunkFood02** [JunkFood02/Seal#93] [Show and tell]
As I mentioned in the release announcement for v1.0.0 (#77), Seal is not even intended to be a video downloader at first. But now, we got a large variety of users and contributors. Thank you!

A month earlier, I received an issue (#27) requesting to downgrade the minSdkVersion to 23. At first, I was a bit surprised because both devices I'm currently using are running the latest version of Android. And I can fully use the dynamic color theme of Material You without any extra codes, a new system feature introduced in API 31.

After some investigation and Ctrl+C/V, dark mode and custom color themes work on devices running even Android 6. Hope you like it! (credits to the compatibility of Android and Jetpack Libraries)

Later, I received another issue regarding "device rotation" (#43). But sooner I found out that the real problem is the size or the DPI of the screen.

In short, we can't assume that all users use **regular-sized screens**, **latest systems**, and **widely spoken languages**. Here are some screenshots showing several corner cases we handle.

<div align="center">
<img src="https://user-images.githubusercontent.com/69683722/182574684-3014a358-d052-410e-9c53-e57cc13e108d.png" width="30%" />
<br>
Screen size: 1280*720, API 23
<br>
<br>
</div>

<div align="center">
<img src="https://user-images.githubusercontent.com/69683722/182575195-c4e859f6-b10f-44f9-960d-d790fb7642e5.png" width="60%" />
<br>
Tablet, Landscape, API 33
<br><br>
</div>

<div align="center">
<img src="https://user-images.githubusercontent.com/69683722/182577967-0a7ce365-d97b-48e8-8547-00c9d4d1dda6.jpg" width="30%" />
<br>
Arabic, RTL layout, mirrored icons 
<br>
<br>
</div>

<div align="center">
<img src="https://user-images.githubusercontent.com/69683722/182578412-41385d45-e47a-489d-b49b-d6438a3091fc.jpg" width="30%" />
<br>
French, the text of the "Start download" button is wrapped into two lines (now fixed)
<br>
<br>
</div>

**Comment by JunkFood02** [comment]
Thus, it's not planned for major UI modifications during the current stage of development. It is for providing the same user experiences for all, from affordable devices to tablets or foldable, from Android 6 to Android 13.
Also, bug reports and requests for minor improvements are always welcome!

**Reply by berlin46** [reply]
If i add a youtube playlist to download,after it start to download the 1st video,and now i want to cancel the whole download (downloading playlist) ,how can be done ? Please add one click cancel operation

**Reply by JunkFood02** [reply]
Are you using the latest version, v1.0.0? If this is the case, you may notice that there's an indicator under the "Seal" app name when downloading a playlist.

**Reply by berlin46** [reply]
The video that got started, doesn't get cancelled.

**Reply by JunkFood02** [reply]
> The video that got started, doesn't get cancelled.

This is a known issue (#63). Currently, we can only cancel a download by force closing the app.

**Reply by yant0** [reply]
I cut off my phone's internet if i wanted to cancel a download
